### PR TITLE
core/internal: add 3-arg newStream method to ClientTransport interface

### PR DESCRIPTION
**Pull request by dapengzhang0**
adding 
`ClientStream newStream(MethodDescriptor<?, ?> method, Metadata headers, CallOptions callOptions);`
to `ClientTransport` interface

Created this PR first because both fail fast implementation and another change will be using this interface change

### CODE REVIEW

File: core/src/test/java/io/grpc/internal/DelayedClientTransportTest.java [hunk 1]
```diff
@@ -96,8 +96,10 @@
 
   @Before public void setUp() {
     MockitoAnnotations.initMocks(this);
-    when(mockRealTransport.newStream(same(method), same(headers))).thenReturn(mockRealStream);
-    when(mockRealTransport2.newStream(same(method2), same(headers2))).thenReturn(mockRealStream2);
+    when(mockRealTransport.newStream(same(method), same(headers), any(CallOptions.class)))
+        .thenReturn(mockRealStream);
```
Comment (zhangkun83) [inline core/src/test/java/io/grpc/internal/DelayedClientTransportTest.java]: We should probably create a `CallOptions`, change every `newStream()` call to the new one that passes in the `CallOptions`, and verify that it's propagated to the transport, like what we are doing for `method` and `headers`.

File: core/src/test/java/io/grpc/internal/ClientCallImplTest.java [hunk 2]
```diff
@@ -156,7 +158,7 @@ public void advertisedEncodingsAreSent() {
     call.start(callListener, new Metadata());
 
     ArgumentCaptor<Metadata> metadataCaptor = ArgumentCaptor.forClass(Metadata.class);
-    verify(transport).newStream(eq(method), metadataCaptor.capture());
+    verify(transport).newStream(eq(method), metadataCaptor.capture(), same(CallOptions.DEFAULT));
```
Comment (zhangkun83) [inline core/src/test/java/io/grpc/internal/ClientCallImplTest.java]: We will need a new test method for the propagation of `CallOptions` to `newStream()`.

File: core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java [hunk 3]
```diff
@@ -200,14 +200,16 @@ public void twoCallsAndGracefulShutdown() {
     when(mockTransportFactory.newClientTransport(
             any(SocketAddress.class), any(String.class), any(String.class)))
         .thenReturn(mockTransport);
-    when(mockTransport.newStream(same(method), same(headers))).thenReturn(mockStream);
+    when(mockTransport.newStream(same(method), same(headers), any(CallOptions.class)))
```
Comment (zhangkun83) [inline core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java]: ditto for testing out `CallOptions` propagation

File: core/src/test/java/io/grpc/internal/ManagedChannelImplTransportManagerTest.java [hunk 4]
```diff
@@ -196,7 +197,8 @@ public void reconnect() throws Exception {
     ClientTransport rt2 = transportInfo.transport;
     // Make the second transport ready
     transportInfo.listener.transportReady();
-    verify(rt2, timeout(1000)).newStream(same(method), any(Metadata.class));
+    verify(rt2, timeout(1000)).newStream(same(method), any(Metadata.class),
+        same(CallOptions.DEFAULT));
```
Comment (zhangkun83) [inline core/src/test/java/io/grpc/internal/ManagedChannelImplTransportManagerTest.java]: ditto for testing out CallOptions propagation

**Comment by dapengzhang0** [comment]
PTAL

**Comment by zhangkun83** [comment]
LGTM Thank you for doing this!

**Pull request merged**